pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -217,10 +217,19 @@ func RunRollingUpdate(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args
 			return cmdutil.UsageErrorf(cmd, "please make sure %s exists and is not empty", filename)
 		}
 
-		switch t := infos[0].AsVersioned().(type) {
+		uncastVersionedObj, err := legacyscheme.Scheme.ConvertToVersion(infos[0].Object, v1.SchemeGroupVersion)
+		if err != nil {
```

Comment:
follow up to make this a more reasonable error if you give this something other than a replication controller